js/modules/loki_public_chat_api.js

```diff
@@ -306,17 +307,37 @@ class LokiPublicChannelAPI {
     };
   }
 
+  // get moderation actions
+  async pollForModerators() {
+    try {
+      await this.pollOnceForModerators();
+    } catch (e) {
+      log.warn(`Error while polling for public chat moderators: ${e}`);
+    }
+    if (this.running) {
+      this.timers.channel = setTimeout(() => {
+        this.pollForModerators();
+      }, PUBLICCHAT_MOD_POLL_EVERY);
+    }
+  }
+
   // get moderator status
-  async refreshModStatus() {
+  async pollOnceForModerators() {
     // get moderator status
-    const res = await this.serverRequest('loki/v1/user_info');
-    // if no problems and we have data
-    if (!res.err && res.response && res.response.data) {
-      this.modStatus = res.response.data.moderator_status;
+    const res = await this.serverRequest(
+      `loki/v1/channel/${this.channelId}/get_moderators`
+    );
+    const ourNumber = textsecure.storage.user.getNumber();
+
+    // Get the list of moderators if no errors occurred
+    const moderators = !res.err && res.response && res.response.moderators;
+
+    // if we encountered problems then we'll keep the old mod status
+    if (moderators) {
+      this.modStatus = moderators.includes(ourNumber);
```

Comment:
are we still u sing modStatus?

Reply:
we use it internally in the api for message deletion stuff.